### PR TITLE
Add "every component" block

### PR DESCRIPTION
**Pull request by Vishwas-Adiga**
This PR adds an "every component" block that returns a list of all components of a given type.
See [this community topic](https://community.appinventor.mit.edu/t/feature-request-any-all-components/29567) for more context.

Screenshots:
image
image

### CODE REVIEW

**Comment by ewpatton** [comment]
@Vishwas-Adiga This is a good start, but since it changes the blocks that are available in the companion, you should rebase the change onto `ucr`, bump the `BLOCKS_LANGUAGE_VERSION` in YaVersion, and add an upgrade path in versioning.js. From a design perspective, what other designs did you consider and why do you think this approach is the best option?

**Comment by Vishwas-Adiga** [comment]
Thanks for the pointers, @ewpatton . I'll make the required changes this weekend

I'll also try and link to a design document outlining how I came to this block design for the required functionality, if such an arrangement works for you?

**Comment by ewpatton** [comment]
@Vishwas-Adiga Yes that's fine.

File: appinventor/blocklyeditor/src/generators/yail/componentblock.js [hunk 1]
```diff
@@ -332,3 +332,17 @@ Blockly.Yail.component_component_block = function() {
...
+
+/**
+ * Returns a function that takes no arguments, generates Yail to get a list of all
+ * components of a type, and returns a 2-element array containing the component
+ * getter code and the operation order Blockly.Yail.ORDER_ATOMIC.
+ *
+ * @param {String} instanceName
+ * @returns {Function} component getter code generation function with instanceName bound in
+ */
+Blockly.Yail.component_all_component_block = function() {
```
Comment (ewpatton) [inline appinventor/blocklyeditor/src/generators/yail/componentblock.js:344]: For forwards compatibility, please switch this to `Blockly.Yail['component_all_component_block']`

**Review by ewpatton — CHANGES_REQUESTED** [review]
Thanks @Vishwas-Adiga. This looks pretty good. Please merge `ucr` to pick up @BeksOmega's changes and there are a few minor things to address, but otherwise this looks pretty good to me.

**Comment by Vishwas-Adiga** [comment]
Sorry, I've been unable to work on this lately. I'll work on the style changes and bring the branch up to date right away

**Review by ewpatton — APPROVED** [review]

**Pull request merged**